### DISCUSSION TITLE
Which library to use?

### DISCUSSION THREAD

**Discussion by gintsmurans** [WalletConnect/walletconnect-monorepo#3655] [Developer Support] [answered]
Hi!

We are trying to make wallet connect to work with the web app we are developing, but came across few issues on ios. So we started exploring a bit more of what is what and it did raise a bit of confusion.

To figure it all out, from the dapp point of view here is a bit of summary:

1. v1 is done, closed and will not work anymore.
2. There are providers like @walletconnect/ethereum-provider
3. Then there is modal from @walletconnect/modal
4. Also there is a project - web3Modal, which have multiple implementations, but from what I gathered only wagmi and ethers will be supported.

Currently we are using `EthereumProvider.init` with `showQrModal: true` which uses @walletconnect/modal, but there is no metamask support and although we can connect metamask by scanning a qr code, it is not working very smoothly. At the moment I am testing it on iPad and its not working at all, at the same time it is working on android. 
Then we see that web3Modal has metamask support, which makes us question which library we should work with as it "feels" like @walletconnect/modal will slowly stop receiving updates. 
On the other hand, we have a bit of complicated setup with our own context for gathering account balances because of various environments the app is compiled for - for example, would web3Modal also work in extension environment thus can be used as one and only provider for connecting users to web3 wallets? If not, can web3Modal be used programmatically, i.e. without using its own context (react app)?

Uff lots of questions...

**Comment by glitch-txs** [comment, marked as answer]
Hi, we're currently developing Web3Modal v3 which currently supports vanilla javascript and will eventually support ethers.js as well
https://docs.walletconnect.com/2.0/web3modal/v3/about
If there's any feature you think is missing/feedback please let me know! Thanks.
